teuthology/orchestra/remote.py

```diff
@@ -518,49 +520,59 @@ def move_file(self, src, dst, sudo=False, mode=None, owner=None,
             args += ' && ' + chown + ' ' + owner + ' ' + dst
         self.run(args=args)
 
+    @contextmanager
+    def _maybe_copy_from(self, sudo, path):
+        try:
+            if sudo:
+                temp_path = self.mktemp()
+                self.sh(f'sudo cp ${path} ${temp_path}')
+                self.sh(f'sudo chmod 0666 ${temp_path}')
+                yield temp_path
+            else:
+                yield path
+        finally:
+            if sudo:
+                self.sh(f'rm ${temp_path}')
+
     def read_file(self, path, sudo=False, stdout=None,
-                              offset=0, length=0):
+                              offset=0, length=-1):
         """
         Read data from remote file
 
         :param path:    file path on remote host
         :param sudo:    use sudo to read the file, defaults False
         :param stdout:  output object, defaults to io.BytesIO()
         :param offset:  number of bytes to skip from the file
-        :param length:  number of bytes to read from the file
-
-        :raises: :class:`FileNotFoundError`: there is no such file by the path
-        :raises: :class:`RuntimeError`:      unexpected error occurred
-
```

Comment:
This is the interface change, requires more testing.

Reply:
i just reverted the interface change you made in https://github.com/ceph/ceph/pull/36211.

Reply:
@tchaikov and this bad because this interface was used in other clients, like cephadm.